src/Product/WCProductAdapter.php

```diff
@@ -654,6 +678,10 @@ public function map_gla_attributes( array $attributes ): WCProductAdapter {
 	 * phpcs:disable WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
 	 */
 	public function setTargetCountry( $targetCountry ) {
+		// remove shipping for current target country
+		$this->remove_shipping_country( $this->getTargetCountry() );
```

Comment:
I'm a bit confused on why this is actually needed. I can see it's needed for the test where we create the adapted product with one target country and then later set another target country.
Although I don't see in which scenario this is done within the code, do we ever change the target country like that?

Reply:
Yeah, we're probably never going to use it because we always set it when constructing the object. But it inherits this method from the parent Google Product class. So there's a chance that third-parties use this method to change the target country code. I just think it'd be safer to have some code place to handle this change and sync other parts of `WCProductAdapter` accordingly.